src/index.js

```diff
@@ -336,6 +342,24 @@ class Collection {
 		});
 	}
 
+	/**
+	 * Get all existing values for a given key across a collection
+	 * @param {ValueObject} valueOption - Value options
+	 * @returns {Promise<T[]>} Array of unique values
+	 */
+	values(valueOption) {
+		if (!valueOption) return Promise.reject("Value option must be provided");
+		if (typeof valueOption.field !== "string") return Promise.reject("Field must be a string");
+		if (valueOption.flatten !== undefined && typeof valueOption.flatten !== "boolean")
+			return Promise.reject("Flatten must be a boolean");
+
+		return this.select({ fields: [valueOption.field] })
+			.then((res) => Object.values(res).map((el) => el[valueOption.field]))
+			.then((values) => (valueOption.flatten ? values.flat() : values))
+			// remove null items (if key doesn't exist) and remove duplicates
+			.then((values) => values.filter((e, i, a) => e != null && a.indexOf(e) === i));
```

Comment:
You can only do primitive fields as indexOf doesn't work with complex data types:
![image](https://github.com/TheRolfFR/firestorm-db/assets/18086786/e810f43f-4090-48d3-bebe-66779740e65d)
You should go with findIndex and a good predicate to remove duplicates

Reply:
I tried out using findIndex with a predicate and it's really really slow, because it has to filter every key by iterating over every property twice (O(n^2)).

I feel like a quick solution that might work is just comparing a JSON stringified versions for duplicates, and the user can just slap on their own filter afterwards if they need to.